test/regex/replaceTricky.good

```diff
@@ -0,0 +1,32 @@
+replacing bb b->a
+aa
+(aa, 2)
+count=0 (bb, 0)
+count=1 (ab, 1)
+count=2 (aa, 2)
+count=3 (aa, 2)
+count=4 (aa, 2)
+replacing bb ''->a
+ababa
+(ababa, 3)
+count=0 (bb, 0)
+count=1 (abb, 1)
+count=2 (abab, 2)
+count=3 (ababa, 3)
+count=4 (ababa, 3)
+replacing bb b|->a
+aa
+(aa, 2)
+count=0 (bb, 0)
+count=1 (ab, 1)
+count=2 (aa, 2)
+count=3 (aa, 2)
+count=4 (aa, 2)
```

Comment:
if I'm understanding correctly, this is a departure from Python's implementation (which your comments indicates behaves differently for `|b` and `b|`. Could you mention this in the OP, and note your rationale?

Reply:
At first I thought it was an RE2 bug / odd behavior. It's true that RE2 is different from Python, but the RE2 behavior isn't what we're getting from this program in Chapel. (It's also not what earlier Chapel releases did).

Here is a C++-only test that shows `b|` producing `aa` but `|b` producing `ababa`, which makes sense to me based on what the RE2 documentation describes (the first option in the `|` is preferred).

<details>

``` c++
#include <re2/re2.h>
#include <iostream>

int main() {
  auto regex = new RE2("b|");
  std::string s;
  s = "bb";
  RE2::GlobalReplace(&s, RE2("b"), "a");
  std::cout << "replace bb b->a :: " << s << "\n";

  s = "bb";
  RE2::GlobalReplace(&s, RE2(""), "a");
  std::cout << "replace bb ''->a :: " << s << "\n";

  s = "bb";
  RE2::GlobalReplace(&s, RE2("b|"), "a");
  std::cout << "replace bb b|->a :: " << s << "\n";

  s = "bb";
  RE2::GlobalReplace(&s, RE2("|b"), "a");
  std::cout << "replace bb |b->a :: " << s << "\n";

  return 0;
}
```

</details>

I'll investigate.

Reply:
The problem here is that the RE2 regex options aren't being set correctly. We set `longest_match=!nonGreedy`; since `nonGreedy` defaults to false, `longest_match` is normally set. I don't think this makes sense, because `nonGreedy` is very different from `longest_match`. I think it's a separate problem from what this PR is doing, however. I'll make an issue.

Reply:
I've created #24788 about this problem.